From 7f00dcb548584911191f1e70483d4523c3ba1319 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Fri, 15 Mar 2013 12:35:36 +0100
Subject: [PATCH] schedwrk: Set values before create callback

When totem callback is called before setting value, NULL can be called,
resulting in segfault.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 exec/schedwrk.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/exec/schedwrk.c b/exec/schedwrk.c
index 0591063..e7ffe92 100644
--- a/exec/schedwrk.c
+++ b/exec/schedwrk.c
@@ -118,6 +118,10 @@ static int schedwrk_internal_create (
 		goto error_destroy;
 	}
 
+	instance->schedwrk_fn = schedwrk_fn;
+	instance->context = context;
+	instance->lock = lock;
+
 	totempg_callback_token_create (
 		&instance->callback_handle,
 		TOTEM_CALLBACK_TOKEN_SENT,
@@ -125,10 +129,6 @@ static int schedwrk_internal_create (
 		schedwrk_do,
 		handle2void (*handle));
 
-	instance->schedwrk_fn = schedwrk_fn;
-	instance->context = context;
-	instance->lock = lock;
-
         hdb_handle_put (&schedwrk_instance_database, *handle);
 
 	return (0);
-- 
1.7.1

